cpp/src/io/orc/reader_impl.cu

```diff
@@ -777,7 +777,7 @@ table_with_metadata reader::impl::read(size_type skip_rows,
             break;
           }
         }
-        out_buffers.emplace_back(column_types[i], num_rows, is_nullable, stream, _mr);
+        out_buffers.emplace_back(column_types[i], num_rows - skip_rows, is_nullable, stream, _mr);
```

Comment:
I don't think this is the right way to fix this. AFAIK `num_rows` should be equal to the number of rows to read. Instead of patching the value here, IMO the fix is to set the correct `num_rows` value in `select_stripes()`.